File: SiliconLabsApp/Models/BluetoothModels/SILBluetoothServiceCharacteristicModel.m
//
//  SILBluetoothServiceCharacteristicModel.m
//  SiliconLabsApp
//
//  Created by Eric Peterson on 10/20/15.
//  Copyright © 2015 SiliconLabs. All rights reserved.
//

#import "SILBluetoothServiceCharacteristicModel.h"

@implementation SILBluetoothServiceCharacteristicModel

- (instancetype)initWithName:(NSString *)name type:(NSString *)type {
    self = [super init];
    if (self) {
        self.name = name;
        self.type = type;
    }
    return self;
}

@end

File: SiliconLabsApp/Models/SILBluetoothServiceCharacteristicModel.swift
//
//  SILBluetoothServiceCharacteristicModel.swift
//  BlueGecko
//
//  Created by Grzegorz Janosz on 15/04/2021.
//  Copyright © 2021 SiliconLabs. All rights reserved.
//

import Foundation

@objcMembers
class SILBluetoothServiceCharacteristicModel : NSObject {
    let name: String
    let type: String
    let properties: SILBluetoothServiceCharacteristicProperties
    var descriptors: [SILBluetoothServiceDescriptorModel]?
    
    init(name: String, type: String, properties: SILBluetoothServiceCharacteristicProperties) {
        self.name = name
        self.type = type
        self.properties = properties
    }
}
